### DISCUSSION TITLE
Course selection

### DISCUSSION THREAD

**Discussion by snatarajappa** [snigi-gupta/AttendingUniversityAtBuffalo#19] [Q&A] [answered]
Hello Snigdha,

Thank you very much for all the information provided here.
I am facing dilemma in selecting courses. 

What are the differences between **CSE 586 - Distributed Systems**  by _Ethan Blanton_ and **CSE 603 - Parallel and Distributed Processing** by _Jaric Zola_? If they are different, is it worth consider taking both courses on same semester and what is the difficulty?

Thanks in advance,

Santhosh

**Comment by snigi-gupta** [comment, marked as answer]
Hi @snatarajappa 

DS is more about CAP theorem, how DSs are made, and how systems interact with each other on a larger level. There are many DS concepts that you will get to learn in this course like linearizability, availability, fault tolerance, etc. 

PDP is about supercomputers, GPUs, and parallel programming. 

There is an in-depth [PDP review here](https://github.com/snigi-gupta/AttendingUniversityAtBuffalo/blob/main/Computer%20Science%20Important%20Tips/CourseReview.md)

Both courses in one sem will definitely be a heavy load to commit to. Both DS and PDP are tough courses by good professors.

**Reply by snatarajappa** [reply]
@snigi-gupta  Thanks a lot.
